modules/utask/execution_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package utask
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func init() {
+	_backend = &InMemBackend{}
+}
+
+func TestEnqueueNonFunction(t *testing.T) {
+	err := Register("I am not a function")
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "Expected a func as input but was")
+}
+
+func TestEnqueueWithMultipleReturnValues(t *testing.T) {
+	fn := func() (string, error) { return "", nil }
+	err := Register(fn)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "Expected function to return only error but found")
```

Comment:
I think too little can fail to check for the right test case since error messages can be similar in wording. Will try to strike a balance.